### ISSUE TITLE
Configuration option to trigger a rebuild if a variable changes

### ISSUE THREAD

**Issue by LukBukkit** [go-task/task#394]
Hello,
I'm using Task version 3.0.0 on Windows and I'm very happy about this build tool.

While using this program, I noticed one feature would be very helpful: Trigger a rebuild based on the change of a variable.
In my case, it's the `PURGE` variable which if activated, generates a smaller file.

If I run at first `task build assets` and then `task build assets PURGE=true`, no rebuild is triggered. But in this case it would be very useful, because the output file would change. Of course you can use the `--force` option to always trigger a rebuild.

I tried to implement my own workaround. But this does only work on Unix shells (/bin/bash, Git Bash ...) and not if you're using the PowerShell.

Here you can take a look at my `Taskfile.yml`
```yml
version: '3'

tasks:
  assets:
    desc: Creates the assets using yarn
    dir: resources
    cmds:
      - yarn
      - yarn run build
      - echo "{{if .PURGE}}true{{else}}false{{end}}" > last_build_mode # my workaround
    sources:
      - styles.css
    generates:
      - output.css
    status:
      - cat last_build_mode
      - test "{{if .PURGE}}true{{else}}false{{end}}" = $(cat last_build_mode) # my workaround
    env:
      NODE_ENV:
        sh: echo {{if .PURGE}} production {{else}} development {{end}}
```

My suggestion would be to add a new field like `var-deps`, where you can specify variables which trigger a rebuild.
```yml
version: '3'

tasks:
  assets:
    desc: Creates the assets using yarn
    dir: resources
    cmds:
      - yarn
      - yarn run build
    sources:
      - styles.css
    generates:
      - output.css
    var-deps: 
      - PURGE
```

Thanks a lot for this useful software. :)

**Comment by andreynering** [comment]
Hi @LukBukkit, thanks for opening this issue!

This sounds like a very valid use case / feature request.

**Comment by andreynering** [comment]
I'd like to note that, specifically for sources when using the checksum mode, there's actually a way of achieving that currently:


```yml
version: '3'

tasks:
  my-task:
    # ...
    label: 'my-task={{.VARIABLE}}'
```

The `label:` prop is used to identify the task, so concatenating a variable there will make it have a different key when that changes.

This will not work for timestamp mode or plain `status:`, though. So this feature request still makes sense.

**Comment by sylv-io** [comment]
There is an alternative workaround by using a helper script for status:
https://github.com/system-transparency/system-transparency/commit/38fd6cbd928d9125bccda4588b733fd73c8e54f3

tl;dr: it creates a config file under `.task/config/<target>` based on a variable name pattern. Any change does trigger a rebuild.

However the script `.task_config.sh` is written specifically for the project, but you could adapt it to generate a sub-config based on your own variable name pattern.

IMHO this could be a great new task feature :+1:  If there is interest, i would give it a try.

**Comment by trulede** [comment]
Closing stale issue.

IIRC this now works as expect with the latest version of Task. Otherwise, please reopen this issue.

**Comment by lukbukkit** [comment]
Thanks!
